app/src/main/java/com/kickstarter/ui/views/CommentCard.kt

```diff
@@ -113,9 +115,11 @@ class CommentCard @JvmOverloads constructor(
             cardCommentStatus == CommentCardStatus.COMMENT_WITHOUT_REPLIES ||
             cardCommentStatus != CommentCardStatus.DELETED_COMMENT
 
-        binding.commentActionGroup.isVisible = cardCommentStatus == CommentCardStatus.COMMENT_WITH_REPLIES ||
-            cardCommentStatus == CommentCardStatus.COMMENT_WITHOUT_REPLIES ||
-            cardCommentStatus == CommentCardStatus.FAILED_TO_SEND_COMMENT
+        if (shouldShowReplyButton(cardCommentStatus)) {
```

Comment:
Question here, why do we need this functionality if we are setting the reply button visibility directly from the view model? I think the  card status should be decoupled from the reply button. This status is representative of showing the actual replies, not the reply button. We are essentially performing this logic twice, when the card status is emitted and when the isReplyButtonVisible stream emits in the view holder.

Reply:
You are completely right! as we spoke on the PR review session earlier we will handle this in a separate PR once @hadia 's work for the retry is completed to avoid her any further conflict 👍

Reply:
About my previous comment, the reply button is tied to the delete status. So the visibility should be updated when changing status
![Screen Shot 2021-06-04 at 11 04 20 AM](https://user-images.githubusercontent.com/4083656/120845292-760da680-c525-11eb-8bf5-6d9a9019dc42.png)

Reply:
Makes sense to me! If that's the case then let's leave it as is 👍🏽